bindings/hnsw_bindings.cpp

```diff
@@ -125,10 +139,14 @@ template <typename dist_t, typename data_t = float> class Index {
   }
 
   void init_new_index(const size_t maxElements, const size_t M,
-                      const size_t efConstruction, const size_t random_seed) {
+                      const size_t efConstruction, const size_t random_seed,
+                      const py::object using_pq) {
```

Comment:
is not only about const, I think it should be a const reference, `const &` to avoid copying at every call

Reply:
`py::object` itself is a reference. The memory management for those objects is held by the Python interpreter.

Reply:
so u mean `py::object` is cheap to copy? Anyway, doing `const &` is better, even if the `py::object` is cheap to copy, there is no point on copying it